packages/clerk-js/src/utils/redirectUrls.ts

```diff
@@ -39,36 +38,69 @@ export class RedirectUrls {
     return this.#getRedirectUrl('signUp');
   }
 
-  toSearchParams() {
-    return this.#toSearchParams(this.#flattenPreservedProps());
+  getPreservedSearchParams() {
+    return this.#toSearchParams(this.#flattenPreserved());
   }
 
-  appendPreservedPropsToUrl(url: string, _otherParams: ParsedQs = {}) {
-    const params = new URLSearchParams();
-    const otherParams = Object.entries(_otherParams);
-    const redirectParams = [...this.#toSearchParams(this.#flattenPreservedProps()).entries()];
-    // merge with existing search params, if any
-    // redirect params should always win
-    [otherParams, redirectParams].flat().forEach(([key, value]) => {
-      typeof value === 'string' && params.set(key, value);
-    });
-
-    // TODO: A potential future improvement here is to remove the origin from the params we append
-    // if `url` and the param share the same origin
-    return buildURL({ base: url, hashSearch: params.toString() }, { stringify: true });
+  toSearchParams() {
+    return this.#toSearchParams(this.#flattenAll());
   }
 
   #toSearchParams(obj: Record<string, string | undefined | null>): URLSearchParams {
     const camelCased = Object.fromEntries(Object.entries(obj).map(([key, value]) => [camelToSnake(key), value]));
     return new URLSearchParams(removeUndefined(camelCased) as Record<string, string>);
   }
 
-  #flattenPreservedProps() {
+  #flattenPreserved() {
     return Object.fromEntries(
       Object.entries({ ...this.fromSearchParams }).filter(([key]) => RedirectUrls.preserved.includes(key)),
     );
   }
 
+  #flattenAll() {
```

Comment:
I decided to stick with a more verbose approach here instead of a a more complex loop of some sort, let me know if you this is is too much

Reply:
I like it